amlip_py/amlip_py/node/FiwareNode.py

```diff
@@ -0,0 +1,509 @@
+
+        """
+
+        self.logger.info(f'Inference received from server: {server_id}\n'
+                         f' with id: {task_id}\n'
+                         f' inference: {inference.to_string()}')
+
+        metadata = {
+            'task_id': {
+                'value': task_id,
+                'type': 'string'
+            },
+            'server_id': {
+                'value': server_id.to_string(),
+                'type': 'string'
+            }
+        }
+
+        self.patch_inference(inference.to_string(), metadata=metadata)
+
+    def patch_inference(
+            self,
+            inference,
+            metadata={}) -> None:
+        """
+        Patch inference data to the Fiware context broker.
+
+        Parameters
+        ----------
+        inference : InferenceSolutionDataType
+            Inference of the data previously sent.
+        metadata : dict
+            Metadata to be included with the inference data.
+            Defaults to an empty dictionary.
+
+        """
+
+        data = {
+            self.entity_solution: {
+                'value': inference,
+                'type': 'InferenceSolutionDataType',
+                'metadata': metadata
+            }
+        }
+
+        try:
+            response = requests.patch(
+                f'http://{self.context_broker_ip}:{self.context_broker_port}/' +
+                f'v2/entities/{self.id}/attrs',
+                headers=headers_POST, data=json.dumps(data))
+
+            response.raise_for_status()
+            self.logger.info('Inference data posted successfully')
+            # __FLAG_
+            # We want to open it again after uploding the inference to the context broker
+            self.waiter_inference.open()
+            # __FLAG__
+
+        except requests.exceptions.RequestException as e:
+            self.logger.warning(f'Failed to patch inference data: {e}')
+            raise e
+
+    def get_inference(
+            self,
+            timeout=0) -> dict:
+        """
+        Get inference data from the Fiware context broker.
+
+        Parameters
+        ----------
+        timeout : int
+            Timeout in seconds for the request.
+            Defaults to 0 (no timeout).
+
+        Returns
+        -------
+        inference : dict
+            Inference data retrieved from the context broker.
+
+        """
+        # __FLAG__
+        inference = {}
+
+        if timeout > 0:
+            reason = self.waiter_inference.wait(timeout)
+            if reason == AwakeReason.timeout:
+                self.logger.warning('Timeout reached while waiting for inference data')
+                inference = json.dumps({'Error': 'Timeout reached' +
+                                        ' while waiting for inference data'})
+                return inference
+            elif reason == AwakeReason.condition_met:
+                pass
+        elif timeout == 0:
+            reason = self.waiter_inference.wait()
```

Comment:
You can call wait() with the timeout argument regardless of its value (0 or otherwise).

Reply:
Done